server/src/main/java/org/opensearch/search/aggregations/bucket/filterrewrite/FilterRewriteOptimizationContext.java

```diff
@@ -99,12 +99,10 @@ void setRanges(Ranges ranges) {
     public boolean tryOptimize(final LeafReaderContext leafCtx, final BiConsumer<Long, Long> incrementDocCount, boolean segmentMatchAll)
         throws IOException {
         segments.incrementAndGet();
-        if (!canOptimize) {
-            return false;
-        }
 
+        if (!canOptimize) return false;
+        if (!segmentMatchAll) return false;
```

Comment:
This code path is common across date histogram aggregations as well? Does this mean we have similar issue there as well?